package/gluon-core/luasrc/lib/gluon/upgrade/021-interface-roles

```diff
@@ -63,4 +63,19 @@ for iface in pairs(interfaces) do
 	end
 end
 
+-- Fix invalid role configurations
+
+uci:foreach('gluon', 'interface', function(interface)
+
+	local function has_role(role)
+		return util.contains(interface.role, role)
+	end
+
+	if has_role('client') and (has_role('mesh') or has_role('uplink')) then
+		-- remove 'client' role
+		util.remove_from_set(interface.role, 'client')
+		uci:set('gluon', interface['.name'], 'role', interface.role)
+	end
+end)
```

Comment:
The exclusivity should also be checked in `check_site.lua` now.

Reply:
Now
````
	interfaces = {
		wan = {
			default_roles = { 'uplink', 'client' }
		}
	}
````
results in
```
*** site.conf error: expected interfaces.wan.default_roles to contain either "client" or "uplink", but not both.
```